### DISCUSSION TITLE
How can we use the GitHub PR reporter?

### DISCUSSION THREAD

**Discussion by eitsupi** [oxsecurity/megalinter#2392] [Q&A] [answered]
Hi, I tried to set up MegaLinter but the comments are not posting in PRQL/prql#1960.
I looked at the website and couldn't figure out how to set it up in detail am I missing something?
Thanks in advance.

**Comment by Kurt-von-Laven** [comment, marked as answer]
Take a look at the error [here](https://github.com/PRQL/prql/actions/runs/4270484674/jobs/7434309891). I've copied the note below the table [here](https://github.com/marketplace/actions/create-or-update-comment#action-inputs) to offer additional context on this error message:

> Note: In _public_ repositories this action does not work in `pull_request` workflows when triggered by forks. Any attempt will be met with the error, `Resource not accessible by integration`. This is due to token restrictions put in place by GitHub Actions. Private repositories can be configured to [enable workflows](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#enabling-workflows-for-forks-of-private-repositories) from forks to run without restriction. See [here](https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#restrictions-on-repository-forks) for further explanation. Alternatively, use the [`pull_request_target`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target) event to comment on pull requests.

**Reply by eitsupi** [reply]
Thank you for your response. Sorry I did not understand the meaning of the error.
I understood that a workflow executed with a PR from a fork cannot create a comment. On my fork the comment was created correctly.
eitsupi/prql#2

**Reply by Kurt-von-Laven** [reply]
No worries; it's a bit cryptic. Yeah, that makes sense since in that case the action isn't triggered by a fork. Does your workflow have the `pull-requests: write` permission?

**Reply by eitsupi** [reply]
Perhaps this is because this workflow definition is only in the fork.
Once merged into the default branch, will it work from the fork?

**Reply by Kurt-von-Laven** [reply]
No, it is because of GitHub Actions' security model. The note I copied above and #2125 go into more detail; let me know if those don't clear things up for you.

**Reply by eitsupi** [reply]
Sorry but I am still confused. Doesn't the documentation you linked to explain private repositories?
If this is mentioned in the MegaLinter documentation that would be great.

**Reply by nvuillam** [reply]
@eitsupi please could you share your .mega-linter.yml config file + your GHA workflow ?

**Reply by Kurt-von-Laven** [reply]
I went into great depth regarding both public and private repositories in the thread I linked.

**Reply by eitsupi** [reply]
> @eitsupi please could you share your .mega-linter.yml config file + your GHA workflow ?

Thank you for your response.
The all code are PRQL/prql#1960 and PRQL/prql#1974.

Note that I just copied the workflow definition from https://megalinter.io/latest/installation/.

I believe I understand that the example shown here does not support pull requests from forks.
I think the user would expect the example to work without doing anything, so either the example or the documentation needs to be updated, correct?

**Reply by Kurt-von-Laven** [reply]
Not opposed to updating MegaLinter's documentation, but to put it in the simplest terms, it sounds like you are trying to do something that GitHub Actions doesn't support.
